ownCloud/Viewer/PDFThumbnailsCollectionViewController.swift

```diff
@@ -0,0 +1,203 @@
+//
+//  PDFThumbnailsCollectionViewController.swift
+//  ownCloud
+//
+//  Created by Michael Neuwert on 04.10.2018.
+//  Copyright © 2018 ownCloud GmbH. All rights reserved.
+//
+
+/*
+ * Copyright (C) 2018, ownCloud GmbH.
+ *
+ * This code is covered by the GNU Public License Version 3.
+ *
+ * For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+ * You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+ *
+ */
+
+import UIKit
+import PDFKit
+import ownCloudSDK
+
+class PDFThumbnailsCollectionViewController: UICollectionViewController, UICollectionViewDataSourcePrefetching, Themeable {
+
+    fileprivate let thumbnailSizeMultiplier: CGFloat = 0.3
+    fileprivate let maxThumbnailCachedCount: UInt = 100
+    fileprivate let verticalInset: CGFloat = 8.0
+    fileprivate let horizontalInset: CGFloat = 4.0
+
+    var pdfDocument: PDFDocument?
+    var themeCollection: ThemeCollection?
+    let thumbnailCache = OCCache<NSString, UIImage>()
+    var thumbnailFetchQueue = OperationQueue()
+    var fetchOperations : [IndexPath : BlockOperation] = [:]
+
+    init() {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .vertical
+        thumbnailCache.countLimit = maxThumbnailCachedCount
+        super.init(collectionViewLayout: layout)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("not implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.collectionView?.prefetchDataSource = self
+
+        // Register cell classes
+        self.collectionView!.register(PDFThumbnailCollectionViewCell.self,
+                                      forCellWithReuseIdentifier: PDFThumbnailCollectionViewCell.identifier)
+
+        if self.themeCollection != nil {
+            self.collectionView!.applyThemeCollection(self.themeCollection!)
```

Comment:
Ideally, this view controller should register as a client with `Theme` itself so it gets updated if the theme changes dynamically (f.ex. following sunrise/sunset in a future version) and a different `ThemeCollection` becomes the active one.

Reply:
@felix-schwarz fixed